pkg/app/piped/executor/ecs/ecs.go

```diff
@@ -415,9 +433,16 @@ func routing(ctx context.Context, in *executor.Input, platformProviderName strin
 		return false
 	}
 
-	if err := client.ModifyListeners(ctx, currListenerArns, routingTrafficCfg); err != nil {
-		in.LogPersister.Errorf("Failed to routing traffic to PRIMARY/CANARY variants: %v", err)
-		return false
+	if len(listenerRules) == 0 {
+		if err := client.ModifyListeners(ctx, currListenerArns, routingTrafficCfg); err != nil {
```

Comment:
Move [these lines](https://github.com/pipe-cd/pipecd/blob/ce26e5b6ae53ad95d6f8b4a3998a2a9a21d743e3/pkg/app/piped/executor/ecs/ecs.go#L412-L416) inside this block because only this line references `currListenerArns`.

Reply:
Thank you for the review, @kentakozuka!

I've made the suggested changes and moved the lines inside the block as per your feedback. 
[bb1b2f3](https://github.com/pipe-cd/pipecd/pull/4540/commits/bb1b2f34d29a6f4b6a4e2cc011b1775569fec1c7)

 I've also updated the documentation to reflect the changes made in this PR.
 [9a2a9c0](https://github.com/pipe-cd/pipecd/pull/4540/commits/9a2a9c09c763930abfefe7983c84b189a43718b9)

Please let me know if there are any other points I should address.